added --read-only option to remove-grant

wvdemeer opened the pull request:
`remove-grant` currently removes the student for the repo, removing all access.

I've added a `--read-only` option, which instead removes write access to the repo. Students can still create and reply to issues etc.

This is useful after a deadline etc.


(I found this code when cleaning up my ghtt. I used it last year but forgot to make a pull request. Could also be made into a separate command is that is better.)

merlijn-sebrechts commented on ghtt/assignment.py:
```diff
@@ -627,8 +630,19 @@ def remove_grant(ctx, yes, students=None, groups=None):
+        else:
+            # Set Read-only repo access
+            for username in [s.username for s in repo.students]:
+                current_permission = g_repo.get_collaborator_permission(username)
+                click.secho("'{}' currently has '{}' access".format(username, current_permission))
+                if current_permission == 'read':
+                    continue
+                click.secho("Setting repo access to read-only for '{}' in '{}'".format(username, repo.name), fg="green")
+                g_repo.add_to_collaborators(username, 'pull')
```
You don't have to check if the student is already read-only, since the `add_to_collaborators` call is idempotent.

```suggestion
                click.secho("Setting repo access to read-only for '{}' in '{}'".format(username, repo.name), fg="green")
                g_repo.add_to_collaborators(username, 'pull')
```

wvdemeer replied:
This was needed just for the feedback to the ghtt user. But it's not really important to know if anything changed or not. So making it simpler is OK.

merlijn-sebrechts reviewed (changes requested):
This is useful, indeed!

Could we add this to the `ghtt grant` command instead? `ghtt grant --read-only`.

merlijn-sebrechts commented:
`grant` is supposed to overwrite existing permissions. But that might not be obvious for the name itself.

My proposal: add the functionality to `grant`, and add an alias for `grant` that is called `set-grant`. + explicitly state in the `help` section that they overwrite any existing permissions of those users.

`modify`, `update` and `change` might be confusing: does it only update existing contributors or also add new contributors? `set` is better in that regard, imo.

wvdemeer commented:
I've made the changes: I've moved everything to the `grant` command.

The `--help` text of `grant` should be clear. For extra "discoverability", I've added a hint to the `remove-grant` command.

I haven't added an alias though. That's possible but adds to much code to be worth the trouble imho. The changes are limited and quite simple now, which is good.

I'm happy with it. Feel free to tweak it further off course.
